launcher/cli/logging.go

```diff
@@ -102,19 +108,30 @@ func setupLogger() {
 var appToAtomicLevel = map[string]zap.AtomicLevel{}
 var appToAtomicLevelLock sync.Mutex
 
-func createLogger(appID string, loggingDef *launcher.LoggingDef, verbosity int, fileSyncer zapcore.WriteSyncer, consoleSyncer zapcore.WriteSyncer) *zap.Logger {
-	fileCore := zapcore.NewNopCore()
-	if fileSyncer != nil {
-		encoderConfig := zap.NewProductionEncoderConfig()
-		fileCore = zapcore.NewCore(zapcore.NewJSONEncoder(encoderConfig), fileSyncer, zap.InfoLevel)
-	}
+func createLogger(appID string, loggingDef *launcher.LoggingDef, verbosity int, fileSyncer zapcore.WriteSyncer, consoleSyncer zapcore.WriteSyncer, format string) *zap.Logger {
 
 	// It's ok for concurrent use here, we assume all logger are created in a single goroutine
 	appToAtomicLevel[appID] = zap.NewAtomicLevelAt(appLoggerLevel(loggingDef.Levels, verbosity))
-	consoleCore := zapcore.NewCore(zapbox.NewEncoder(verbosity), consoleSyncer, appToAtomicLevel[appID])
+
+	var consoleCore zapcore.Core
+	switch format {
+	case "json":
+		encoderConfig := zapdriver.NewProductionEncoderConfig()
+		consoleCore = zapcore.NewCore(zapcore.NewJSONEncoder(encoderConfig), consoleSyncer, appToAtomicLevel[appID])
```

Comment:
We should probably not use the standard `appToLevel`, `INFO` is probably a better default.

Reply:
I prefer to not make `log-format=json` change the handling of log-levels. adding `-vv` to all our production deployments is good enough.

Reply:
The reality is that I think `log-format=json` is not enough to convey the meaning of the change the flags implies. The thing is that this is more than just changing the output to a JSON format. It connects to Google Stackdriver to upload, so it's more than just spit everything as JSON. 

First, we are missing some `WrapCore` that we had previously:

```
		opts = append(opts, zapdriver.WrapCore(
			zapdriver.ReportAllErrors(true),
			zapdriver.ServiceName(serviceName),
		))
```

I kind of agree indeed about the `-vv`, but I still have doubts, mainly that the `-vvv` could change without "thinking" about the other option.

Let's discuss this in person.